diff --git a/packages/core/src/components/features/account/LoginPage/styles/LoginPage.style.js b/packages/core/src/components/features/account/LoginPage/styles/LoginPage.style.js
index 7bf0989f..a87347db 100644
--- a/packages/core/src/components/features/account/LoginPage/styles/LoginPage.style.js
+++ b/packages/core/src/components/features/account/LoginPage/styles/LoginPage.style.js
@@ -4,7 +4,7 @@
 
 import styled from 'styled-components';
 
-export default styled.div`
+export default styled.div `
   background-color: #d3d3d3;
   padding: 20px;
   margin-top: 30px;
@@ -18,4 +18,8 @@ export default styled.div`
     padding: 10px;
     border: 1px solid #d3d3d3;
   }
-`;
+
+  .CheckBox__input {
+    padding: 0;
+  }
+`;
\ No newline at end of file
diff --git a/packages/core/src/components/features/account/LoginPage/views/LoginPage.view.jsx b/packages/core/src/components/features/account/LoginPage/views/LoginPage.view.jsx
index 96b0e000..2de46531 100644
--- a/packages/core/src/components/features/account/LoginPage/views/LoginPage.view.jsx
+++ b/packages/core/src/components/features/account/LoginPage/views/LoginPage.view.jsx
@@ -5,6 +5,7 @@
 import React from 'react';
 import LoginPageStyle from '../styles/LoginPage.style';
 import Button from '../../../../common/atoms/Button';
+import Checkbox from '../../../../common/atoms/InputCheckbox';
 import Grid from '../../../../common/molecules/Grid';
 import Row from '../../../../common/atoms/Row';
 import Col from '../../../../common/atoms/Col';
@@ -23,6 +24,7 @@ class LoginView extends React.PureComponent<Props> {
     this.state = {
       email: '',
       password: '',
+      rememberUserChecked: true
     };
   }
 
@@ -34,9 +36,15 @@ class LoginView extends React.PureComponent<Props> {
   changeHandler = e => {
     this.setState({
       [e.target.name]: e.target.value,
-    });
+    });  
   };
 
+  handleRememberMe = e => {
+    this.setState({
+      rememberUserChecked: !this.state.rememberUserChecked
+    })
+  }
+
   onFormSubmit = e => {
     e.preventDefault();
     const { onSubmit } = this.props;
@@ -65,7 +73,15 @@ class LoginView extends React.PureComponent<Props> {
   /* eslint-disable */
   render() {
     const { loginInfo } = this.props;
-
+    const inputCheckboxProps = {
+      input: {
+        name: "rememberMe-checkbox",
+        value: this.state.rememberUserChecked,
+        onClick : this.handleRememberMe,
+        className: "rememberme-checkbox"
+      }
+    };
+    console.log("state - ", this.state)
     if (Object.keys(loginInfo).length === 0) {
       return null;
     }
@@ -130,6 +146,23 @@ class LoginView extends React.PureComponent<Props> {
                         />
                       </Col>
                     </Row>
+                    <Row className="marginBottom">
+                      <Col
+                        colSize={{
+                          large: 12,
+                          medium: 12,
+                          small: 12,
+                        }}
+                      >
+                        <Checkbox {...inputCheckboxProps}>
+                          <div>
+                            Remember me.
+                            <div>Not recommended on shared devices.</div>
+                          </div>
+                        </Checkbox>
+                        
+                      </Col>
+                    </Row>
                     <Row className="marginBottom">
                       <Col
                         colSize={{
